### PR TITLE
[1.7.x] Fixed 24352 -- Coercing ManyRelatedManager to string raises StopIteracion exception. 

### PR DESCRIPTION
**Pull request by gmunumel**
Fixed Coercing ManyRelatedManager to string raises Stopteration exception

### CODE REVIEW

**Comment by gmunumel** [comment]
Is isort failed error related to my change?

File: tests/foreign_object/tests.py [hunk 1]
```diff
@@ -365,6 +365,10 @@ def test_many_to_many_related_query_name(self):
         with self.assertRaises(FieldError):
             Article.objects.filter(ideas__name="idea1")
 
+    def test_ticket_24352(self):
```
Comment (timgraham) [inline tests/foreign_object/tests.py]: You can use a more descriptive name like `test_many_related_manager_str`. I think a better place for this test would be `m2m_regress`. Currently, your test is in a class called `MultiColumnFKTests` which isn't relevant to this change.

**Comment by timgraham** [comment]
Please open a new pull request against master as Simon indicated on the ticket, thanks.

**Comment by timgraham** [comment]
Reopening as the bug only affects 1.7. Please update the test per my comments and add a mention in the 1.7.6 release notes, thanks.

**Comment by gmunumel** [comment]
Should I do the PR againt `1.7` branch?

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
merged in 0831a43c3afc3f93b8ed0c58630836e38248be12, thanks.

**Pull request closed without merging**